code/datums/datumvars.dm

```diff
@@ -592,6 +593,30 @@ body
 			if(usr.client)
 				usr.client.cmd_assume_direct_control(M)
 
+		else if(href_list["offer_control"])
+			if(!check_rights(0))	return
+
+			var/mob/M = locate(href_list["offer_control"])
+			if(!istype(M))
+				usr << "This can only be used on instances of type /mob"
+				return
+			M << "Control of your mob has been offered to dead players."
+			log_admin("[key_name(usr)] has offered control of [M.real_name] to ghosts.")
+			message_admins("[key_name(usr)] has offered control of [M.real_name] to ghosts")
+			var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [M.real_name]?", "pAI", null, FALSE, 100)
+			var/mob/dead/observer/theghost = null
+
+			if(candidates.len)
+				theghost = pick(candidates)
+				M << "Your mob has been taken over by a ghost!"
+				usr << "[theghost.key] has taken over [M.real_name]."
+				message_admins("[key_name(theghost)] has taken control of [M.real_name]")
+				M.ghostize()
+				M.key = theghost.key
+			else
+				M << "There were no ghosts willing to take control."
+				message_admins("No ghosts were willing to take control of [M.real_name]")

```

Comment:
As dunc said, this is used to take over mobs, dunno why you'd want it to display the key name if you, for example, spawned a morph and then offered control to ghosts.